src/file.cpp

```diff
@@ -509,7 +509,14 @@ void SyncFileGet::updatelocalname()
 // add corresponding LocalNode (by path), then self-destruct
 void SyncFileGet::completed(Transfer*, LocalNode*)
 {
-    sync->checkpath(NULL, &localname);
+    LocalNode *ll = sync->checkpath(NULL, &localname);
```

Comment:
Review the case when there were an existing localnode before, pointing to the same file. If so, check the former outdated localnode is deleted accordingly.